algosdk/abi/interface.py

```diff
@@ -0,0 +1,39 @@
+import json
+
+from algosdk.abi.method import Method
+
+
+class Interface:
+    """
+    Represents a ABI interface description.
+
+    Args:
+        name (string): name of the interface
+        methods (list): list of Method objects
+    """
+
+    def __init__(self, name, methods) -> None:
+        self.name = name
+        self.methods = methods
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, Interface):
+            return False
+        return self.name == o.name and self.methods == o.methods
```

Comment:
```suggestion
        return self.id == o.id or (self.name == o.name and self.methods == o.methods)
```
Not really required, just a handy trick that speeds up evaluation `x == x` (especially for nested structures). But it depends on the usage pattern as it does add an extra evaluation in other cases.